tests/lib/Service/OpenProjectAPIServiceTest.php

```diff
@@ -1183,11 +1202,16 @@ public function testGetOpenProjectAvatar() {
 			->setHeaders(["Authorization" => "Bearer 1234567890"]);
 
 		$providerResponse = new ProviderResponse();
+
 		$providerResponse
 			->setStatus(Http::STATUS_OK)
 			->setHeaders(['Content-Type' => 'image/jpeg'])
-			//setBody() expects iterable but we want to have raw data here and it seems to work fine
-			->setBody('dataOfTheImage');
+			->setBody(
+				new Binary(
+					__DIR__ . "/../fixtures/openproject-icon.jpg",
+					'image/jpeg'
+				)
+			);
```

Comment:
```suggestion
			->setBody(
				new Binary(
					__DIR__ . "/../fixtures/openproject-icon.jpg",
					'image/jpeg'
				)
			);
```

there is a problem with content-type having lowercase, since from openproject side the response we get will be upper case `Content-Type` headers key

Reply:
`$response->getHeader('Content-Type')` will get the header correctly, the cases do not matter anymore 
see https://github.com/nextcloud/integration_openproject/pull/666/files#diff-acd9bb903dbec989fbe06b290cfcaed634a3f67d497845d4d6946fa1b6449c93R324
and also my comment in the original post